docs/source/how-to-upstream.mdx

```diff
@@ -1,31 +1,61 @@
 # Upload files to the Hub
 
-Sharing your files and work is a very important aspect of the Hub. The `huggingface_hub` uses a Git-based workflow to upload files to the Hub. You can use these functions independently or integrate them into your own library, making it more convenient for your users to interact with the Hub. This guide will show you how to:
+Sharing your files and work is a very important aspect of the Hub. The `huggingface_hub`
+uses a Git-based workflow to upload files to the Hub. You can use these functions
+independently or integrate them into your own library, making it more convenient for
+your users to interact with the Hub. This guide will show you how to:
 
+* Push a single file to the Hub with the [`upload_file`] function.
 * Push files with a `commit` context manager.
 * Push files with the [`~Repository.push_to_hub`] function.
-* Upload very large files with [Git LFS](https://git-lfs.github.com/).
+* Push very large files with [Git LFS](https://git-lfs.github.com/).
 
-Whenever you want to upload files to the Hub, you need to log in to your Hugging Face account:
+Refer to the following chart to select a uploading method that works best for you and
+then read more to learn about it.
+
+![Hub upload
+flowchart](https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub_upload_flowchart.svg)
+
+Whenever you want to upload files to the Hub, you need to log in to your Hugging Face
+account:
 
 1. Log in to your Hugging Face account with the following command:
 
    ```bash
    huggingface-cli login
    ```
 
-2. Alternatively, if you prefer working from a Jupyter or Colaboratory notebook, login with [`notebook_login`]:
+2. Alternatively, if you prefer working from a Jupyter or Colaboratory notebook, login
+   with [`notebook_login`]:
 
    ```python
    >>> from huggingface_hub import notebook_login
    >>> notebook_login()
    ```
 
-   [`notebook_login`] will launch a widget in your notebook from which you can enter your Hugging Face credentials.
+   [`notebook_login`] will launch a widget in your notebook from which you can enter
+   your Hugging Face credentials.
+
+## Push without Git commands
+
+The [`upload_file`] function uploads a local file to a repository without using Git
+commands. You need to provide the path of the local file, the desired path of the file
+in the repository, and the repository you want to upload the file to.
+
+```py
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+>>> api.upload_file(path_or_fileobj="/local/path/to/README.md",
+...                 path_in_repo="README.md",
+...                 repo="lysandre/test-model"
+... )
+```
```

Comment:
Thanks, good to know! Since `upload_file` is going to be deprecated soon, I think it'll be better not to introduce this change in the docs right now.

What will be the new recommended way to upload a file, the commit context manager? We will need to update the quickstart as well which also recommends using `upload_file`.